### PR TITLE
feat(tools:add): add new tool named KMD HMDB data plot v1.0.0

### PR DESCRIPTION
**Pull request by Lain-inrae**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by Lain-inrae** [comment]
Est-ce que quelqu'un peut review ma MR, s'youplait? @coreteam

**Comment by Lain-inrae** [comment]
Can you review my PR please..? @workflow4metabolomics/coreteam

File: tools/kmd_hmdb_data_plot/macro.xml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        m/zRatio±MassTolerance and we will plot those
+        compounds.
+        Choose the mass tolerance according to this information.
+      "
+      label="M/z Ratio"
+    />
+
+    <param argument="--mass-tolerance" type="float" value="@DEFAULT_TOLERENCE@"
+      help="
+        The database will give us all compounds with m/z =
```
Comment (bgruening) [inline tools/kmd_hmdb_data_plot/macro.xml:21]: I really wonder how those multi-line help sections look like :)

File: tools/kmd_hmdb_data_plot/macro.xml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    <data name="output_path" format="csv"
+      label="csv - ${tool.name} on ${what.mz_ratio}±${what.mass_tolerance} - ${what.database}"
+    >
+      <filter>"get_data" in str(what['to_do'])</filter>
+    </data>
+  </xml>
+
+  <xml name="produce_plot_outputs">
+    <data name="output" format="html"
+      label="html on ${
```
Comment (bgruening) [inline tools/kmd_hmdb_data_plot/macro.xml:152]: I would not do those acrobatics on output names. Those informations are all captured inside Galaxy and you can look at them via the GUI.

**Review by bgruening — COMMENTED** [review]
Thanks a lot. The tool is in a pretty good shape. I added a few comments. 

The biggest blocker is that the tests are missing.

**Comment by bgruening** [comment]
@Lain-inrae your last commit has add tools/kmd_hmdb_data_plot/chopin_kmd_hmdb_api_client and a lot of stuff. This was probably not intended. Was it?

**Comment by Lain-inrae** [comment]
> @Lain-inrae your last commit has add tools/kmd_hmdb_data_plot/chopin_kmd_hmdb_api_client and a lot of stuff. This was probably not intended. Was it?

Hello!

It was intended - I have forgotten to add the actual client. The tool cannot work without it.

Those files were generated by an openapi tool (https://github.com/openapi-generators/openapi-python-client), and then I modified some of them to fix a little issue. So those files are part of the tool. They do connection handling, data structures checking and data transformation.

I hope it is not a problem..?

**Comment by Lain-inrae** [comment]
I understand your concerns, and I agree - a packaging would be better.
Unfortunately, my contract is coming to an end.
Packaging it for pypi and then for conda will take too long (I had to wait several months for the conda recipe of w4mrutils).
The amount of code remains acceptable, I think (3Ko to transfert - 2.8K sloc).

**Comment by bgruening** [comment]
Submit it to bioconda, or ping me. I can speed up the process I think. Is that ok?

File: tools/kmd_hmdb_data_plot/kmd_hmdb_data_plot.xml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        <edam_operation>operation_3803</edam_operation>
+    </edam_operations>
+    <requirements>
+        <requirement type="package" version="3.10">python</requirement>
+        <requirement type="package" version="23.1.0">attrs</requirement>
+        <requirement type="package" version="8.1.3">click</requirement>
+        <requirement type="package" version="0.24.1">httpx</requirement>
+        <requirement type="package" version="2.0.3">pandas</requirement>
+        <requirement type="package" version="5.15.0">plotly</requirement>
+        <requirement type="package" version="1.0.1">kmd_hmdb_api_client</requirement>
```
Comment (bgruening) [inline tools/kmd_hmdb_data_plot/kmd_hmdb_data_plot.xml:23]:
```suggestion
        <requirement type="package" version="1.0.1">kmd_hmdb_api_client</requirement>
```

File: tools/kmd_hmdb_data_plot/kmd_hmdb_data_plot.xml [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+]]>
+    </help>
+    <citations></citations>
+    <creator>
+        <person
+            honorificPrefix="Mx."
+            givenName="Lain"
+            familyName="Pavot"
+            jobTitle="Computer-Grrrl"
```
Comment (bgruening) [inline tools/kmd_hmdb_data_plot/kmd_hmdb_data_plot.xml:174]: Is that your title ;)

**Comment by bgruening** [comment]
The tests are failing unfortunately :(

**Comment by Lain-inrae** [comment]
> The tests are failing unfortunately :(

Yeah... 

> Job in error state.. tool_id: kmd_hmdb_data_plot, exit_code: 125, stderr: docker: invalid spec: ::rw: empty section between colons.
See 'docker run --help'.

Seems like a galaxy issue - one of those produce a `-v ::rw` :

> -v "$_GALAXY_JOB_TMP_DIR:$_GALAXY_JOB_TMP_DIR:rw" -v "$TMPDIR:$TMPDIR:rw" -v "$TMP:$TMP:rw" -v "$TEMP:$TEMP:rw" -v "$_GALAXY_JOB_HOME_DIR:$_GALAXY_JOB_HOME_DIR:rw"

**Comment by bgruening** [comment]
Yeah, its fixed in a newer release. I think the latest Docker release broke some semantics.

**Review by bgruening — APPROVED** [review]
Super cool! Thanks a lot @Lain-inrae

**Comment by Lain-inrae** [comment]
Thank you @bgruening !

**Pull request merged**